test/unit/org/apache/cassandra/metrics/CounterMutationMetricsTest.java

```diff
@@ -0,0 +1,174 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.metrics;
+
+import java.util.ArrayList;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import org.apache.cassandra.SchemaLoader;
+import org.apache.cassandra.Util;
+import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.db.ColumnFamilyStore;
+import org.apache.cassandra.db.ConsistencyLevel;
+import org.apache.cassandra.db.CounterMutation;
+import org.apache.cassandra.db.Keyspace;
+import org.apache.cassandra.db.Mutation;
+import org.apache.cassandra.db.RowUpdateBuilder;
+import org.apache.cassandra.exceptions.ConfigurationException;
+import org.apache.cassandra.exceptions.WriteTimeoutException;
+import org.apache.cassandra.schema.KeyspaceParams;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class CounterMutationMetricsTest {
+    private static final String KEYSPACE1 = "CounterMutationTest";
+    private static final String CF1 = "Counter1";
+    private static final String CF2 = "Counter2";
+
+    @BeforeClass
+    public static void defineSchema() throws ConfigurationException
+    {
+        SchemaLoader.prepareServer();
+        SchemaLoader.createKeyspace(KEYSPACE1,
+                                    KeyspaceParams.simple(1),
+                                    SchemaLoader.counterCFMD(KEYSPACE1, CF1),
+                                    SchemaLoader.counterCFMD(KEYSPACE1, CF2));
+    }
+
+    @Test
+    public void testLocksDuplicate()
+    {
+        AssertMetrics metrics = new AssertMetrics();
+
+        ColumnFamilyStore cfsOne = Keyspace.open(KEYSPACE1).getColumnFamilyStore(CF1);
+        ColumnFamilyStore cfsTwo = Keyspace.open(KEYSPACE1).getColumnFamilyStore(CF2);
+
+        cfsOne.truncateBlocking();
+        cfsTwo.truncateBlocking();
+
+        Mutation.PartitionUpdateCollector batch = new Mutation.PartitionUpdateCollector(KEYSPACE1, Util.dk("key1"));
+        batch.add(new RowUpdateBuilder(cfsOne.metadata(), 5, "key1")
+                  .clustering("cc")
+                  .add("val", 1L)
+                  .add("val2", -1L)
+                  .build().getPartitionUpdate(cfsOne.metadata()));
+
+        batch.add(new RowUpdateBuilder(cfsTwo.metadata(), 5, "key1")
+                  .clustering("cc")
+                  .add("val", 2L)
+                  .add("val2", -2L)
+                  .build().getPartitionUpdate(cfsTwo.metadata()));
+
+        new CounterMutation(batch.build(), ConsistencyLevel.ONE).apply();
+
+        metrics.assertLockTimeout(0);
+        metrics.assertLocksPerUpdate(1);
+    }
+
+    @Test
+    public void testLockTimeout() throws Throwable
+    {
+        AssertMetrics metrics = new AssertMetrics(11L);
```

Comment:
consider using a factory method so that the scenario is more readable, e.g:
```
AssertMetrics metrics = AssertMetrics.withCounterWriteRpcTimeoutNanos(11)
```

Reply:
Unfortunately it is not supported by Java to have static method in the inner class.
It's possible to put the static method outside the class, which is not the greatest solution. I am doing another readability improvement instead, but not as great as the suggestion.